### ISSUE TITLE
wattr w / h (and wmv) on KDE Plasma?

### ISSUE THREAD

**Issue by erwin** [I-LeCorbeau/sxutils#1]
![image](https://user-images.githubusercontent.com/11534217/133223442-099c35be-8310-41a0-b357-c1808aeb34f5.png)
(window in the screenshot is not at the top left, but being reported as x=0, y=0...

Not sure how much time you want to spend on other window managers, but it seems that at least on KDE the `wattr` method of checking the widow x and y position isn't working.

I've been looking for a reliable way to check this across window managers to implement some keyboard movement functions, but I haven't found a reliable way yet.

At least on KDE I think you have to get the window frame size first.

Once I have something that works on KDE I'll try to get it working on other window managers.

You might want to see:

https://unix.stackexchange.com/questions/14159/how-do-i-find-the-window-dimensions-and-position-accurately-including-decoration

**Comment by IanLeCorbeau** [comment]
Yeah, the issue comes from the fact that Plasma's KWin is a reparenting window manager, which will cause a bunch of problems with tools that communicate directly with the X server. Reporting the x and y positions as 0 is the most common.

I believe using EWMH might be a way to circumvent that problem. This might be of help: https://stackoverflow.com/questions/12775136/get-window-position-and-size-in-python-with-xlib It's about python, but highlights the problem and gives some solutions that can be converted to C.
